shogun-lib/src/main/java/de/terrestris/shogun/lib/envers/CustomAnnotationRevisionMetadata.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Modified work - Copyright 2012-2020 the original author or authors.
+ * See https://github.com/spring-projects/spring-data-envers/
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package de.terrestris.shogun.lib.envers;
+
+import org.springframework.data.history.RevisionMetadata;
+import org.springframework.data.util.AnnotationDetectionFieldCallback;
+import org.springframework.data.util.Lazy;
+import org.springframework.util.Assert;
+import org.springframework.util.ReflectionUtils;
+
+import java.lang.annotation.Annotation;
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+import java.util.Date;
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * A {@link RevisionMetadata} implementation that inspects the given object for fields with the configured annotations
+ * and returns the field's values on calls to {@link #getRevisionInstant()} and
+ * {@link #getRevisionNumber()}.
+ *
+ * @author Oliver Gierke
+ * @author Jens Schauder
+ */
+public class CustomAnnotationRevisionMetadata<N extends Number & Comparable<N>> implements RevisionMetadata<N> {
```

Comment:
Maybe `ShogunAnnotationRevisionMetadata` is more appropriate here since it is the custom implementation for SHOGun?

Reply:
done